error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -541,6 +541,145 @@ Mono<T> after(Mono<T> mono) {
     }
   }
 
+  /**
+   * Don't unnecessarily transform a {@link Flux#using(Callable, Function)} to a mono, instead use
+   * the equivalent API provided by {@link Mono}.
+   */
+  static final class MonoUsing<
+      D extends AutoCloseable, T, P extends Publisher<? extends T>, M extends Mono<? extends T>> {
+    @BeforeTemplate
+    Mono<T> before(Callable<D> resourceSupplier, Function<D, P> sourceSupplier) {
+      return Flux.using(resourceSupplier, sourceSupplier).single();
```

Comment:
_(Here & below)_
This refaster rule will cause a compiler error if the publisher in the supplier is not of type Mono. 😬 

(Need to think how I can improve the matching here)

Reply:
Introduced a before template matcher [here](https://github.com/PicnicSupermarket/error-prone-support/pull/1393/commits/881cc12e9a8fded6fa34da8de6b7e615eb5ccdb5).